File: exo2_5.py
import os
import sys
import time
import random
debut = time.time()
rand1 = random.randint(1, 10)
rand2 = random.randint(1, 10)
pid_f1 = os.fork()
if pid_f1 == 0:
    time.sleep(rand1)
    print("Fils 1: ", time.time() - debut)
else:
    pid_f2 = os.fork()
    if pid_f2 == 0:
        time.sleep(rand2)
        print("Fils 2: ", time.time() - debut)
    else:
        os.wait()
    os.wait()

File: exo2_5.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
#include <time.h>

int main ()
{
	time_t debut = time(NULL);
	srand(time(NULL));

	for (int i = 1; i < 2; i++) {
		if (fork() == 0) {
			int slip = (rand() % 10) + 1;
			printf ("Fils %d: dodo de %ds\n", i, slip);
			sleep(slip);
			exit(0);
		}
	}

	wait(NULL);
	wait(NULL);

	time_t fin = time(NULL);
	printf ("Durée total : %ld secondes\n", fin - debut);
	return 0;
}
